File: 1122.py
k1, k2 = map(int, input().split())
l1, l2 = map(int, input().split())
 
if (k2-l2) == 0 and abs(k1-l1) == 1 or (k1-l1) == 0 and abs(k2-l2) == 1 or abs(k1-l1) == 1 and abs(k2-l2) == 1:
    print("YES")
else: 
    print("NO")

File: 1122.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
 
int main(){
     
    int x1, y1, x2, y2;
    scanf("%d %d %d %d", &x1, &y1, &x2, &y2);
     
    if((y1-y2)==0 && abs(x1-x2)==1 || (x1-x2)==0 && abs(y1-y2)==1 || abs(x1-x2)==1 && abs(y1-y2)==1){
        printf("YES");
    }else printf("NO");
 
    return 0;
}
